odata_parser/src/main/scala/com/sdl/odata/parser/LiteralsParser.scala

```diff
@@ -45,7 +45,7 @@ trait LiteralsParser extends RegexParsers {
 
   def stringLiteral: Parser[StringLiteral] = stringValue ^^ StringLiteral
 
-  def stringValue: Parser[String] = "'" ~> """([^']|(''))*""".r <~ "'" ^^ { case s => s.replaceAll("''", "'")}
+  def stringValue: Parser[String] = "'" ~> """[^']*(?:''[^']*)*""".r <~ "'" ^^ { case s => s.replaceAll("''", "'")}
```

Comment:
Unfoutunately no, since it causes StackOverflowError which is out of JVM control. This is too specific here.
I can add the long query to the test but it won't show anything we are able to have in the future.